app/src/main/java/com/vitorpamplona/amethyst/ui/screen/ChatroomListFeedView.kt

```diff
@@ -77,19 +87,85 @@ private fun FeedLoaded(
 
-    LazyColumn(
-        contentPadding = PaddingValues(
-            top = 10.dp,
-            bottom = 10.dp
-        ),
-        state = listState
-    ) {
-        itemsIndexed(state.feed.value, key = { index, item -> if (index == 0) index else item.idHex }) { index, item ->
-            ChatroomCompose(
-                item,
-                accountViewModel = accountViewModel,
-                navController = navController
-            )
+    var markedAllAsRead by remember { mutableStateOf<Boolean>(false) }
+
+    Column {
+        TextButton(
+            modifier = Modifier.fillMaxWidth(),
+            shape = MaterialTheme.shapes.large,
+            onClick = { markedAllAsRead = true }
+        ) {
+            Text(stringResource(R.string.mark_all_as_read))
+        }
+
+        LazyColumn(
+            contentPadding = PaddingValues(
+                top = 10.dp,
+                bottom = 10.dp
+            ),
+            state = listState
+        ) {
+            itemsIndexed(state.feed.value, key = { index, item -> if (index == 0) index else item.idHex }) { index, item ->
+                ChatroomCompose(
+                    item,
+                    accountViewModel = accountViewModel,
+                    navController = navController
+                )
+            }
+        }
+    }
+
+    @Composable
+    fun markAllAsRead() {
+        val accountState by accountViewModel.accountLiveData.observeAsState()
+        val account = accountState?.account ?: return
+
+        val notificationCacheState = NotificationCache.live.observeAsState()
+        val notificationCache = notificationCacheState.value ?: return
+
+        val context = LocalContext.current.applicationContext
+
+        state.feed.value.forEach { baseNote ->
+            val noteState by baseNote.live().metadata.observeAsState()
+            val note = noteState?.note
+
+            val routeForLastRead = if (note?.event == null) {
+                return@forEach
+            } else if (note.channel != null) {
+                val channelState by note.channel!!.live.observeAsState()
+                val channel = channelState?.channel
+
+                channel?.let {
+                    "Channel/${it.idHex}"
+                }
+
+            } else {
+                val replyAuthorBase = note.mentions?.first()
+
+                var userToComposeOn = note.author!!
+
+                if (replyAuthorBase != null) {
+                    if (note.author == account.userProfile()) {
+                        userToComposeOn = replyAuthorBase
+                    }
+                }
+
+                "Room/${userToComposeOn.pubkeyHex}"
+            }
+
+            routeForLastRead?.let {
+                LaunchedEffect(key1 = notificationCache) {
+                    val createdAt = note.event?.createdAt
+                    if (createdAt != null) {
+                        NotificationCache.markAsRead(it, createdAt, context)
```

Comment:
Should we consider marking as read with the current time instead of created at?

Reply:
No, because you could see a new message later which has a createdAt time earlier than that timestamp, but you would not want to mark that one as read yet.

Reply:
The same issue applies regardless of if we mark with the current time or with created at.